tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
@xiang90 my earliest code was watching over channels only, but the problem with ranging over channels is it will hanng there only since the channel will not have closed by then. What we can do is add all the watches in a slice and check with ok to ensure its closed. What do you suggest?